aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,198 @@
+
+When a client can process compressed content, the client must inform the server of its capabilities by sending the `Accept-Encoding` header with the request. When a server sends compressed content, it must include information in the `Content-Encoding` header on how the compressed response is encoded. Content encoding designations are shown below indicating which ones are supported by the middleware.
+
+Content Encoding | Supported | Description
+| :---: | :---: | ---
+`br` |  No | Brotli Compressed Data Format
+`compress` | No | UNIX "compress" data format
+custom | Yes | Developer provides the compression implementation. The client must be able to decompress the payload.
+`deflate` |  No | "deflate" compressed data inside the "zlib" data format 
+`exi` | No | W3C Efficient XML Interchange
+`gzip` | Yes (default) | GZip file format
+`identity` | Yes | "No encoding" identifier: The response must not be encoded.
+`pack200-gzip` | No | Network Transfer Format for Java Archives
+`*` | Yes | Any available content encoding not explicitly requested
+
+For more information, see the [IANA Official Content Coding List](http://www.iana.org/assignments/http-parameters/http-parameters.xml#http-content-coding-registry).
+
+The middleware is capable of reacting to quality value (qvalue, `q`) weighting when sent by the client. For more information, see [RFC 7231: Accept-Encoding](https://tools.ietf.org/html/rfc7231#section-5.3.4).
+
+Most compression algorithms are subject to a tradeoff between compression speed and the effectiveness of the compression. *Effectiveness* in this context means the smallest possible size, which is also referred to as the most *optimal* compression. The middleware defaults to the fastest compression level, which might not produce the most efficient compression. If the most efficient compression is desired, the middleware can be configured for optimal compression.
+
+The headers involved in requesting, sending, caching, and receiving compressed content are described below.
+
+Header | Role
+--- | ---
+`Accept-Encoding` | Sent by the client to the server to indicate which types of content encoding are acceptable.
+`Content-Type` | Specifies the MIME type of the content. MIME types are specified on the server for encodable content, so this information is used by the server to determine if a response can be compressed. The middleware includes a set of default MIME types that it will encode, but you can replace or add MIME types.
+`Content-Encoding` | Sent from the server to the client to indicate the encoding of the content in the payload.
+`Vary` | When sent by the server with a value of `Accept-Encoding` to clients and proxies, it indicates that they should cache (vary) responses based on the value of the `Accept-Encoding` header of the request. The result of returning content with the `Vary: Accept-Encoding` header is that both compressed and uncompressed responses will be cached.
+
+## Response compression sample application
```

Comment:
You're correct ... there is no necessity for it as a separate section. I'll kill it here and probably have the text (or some of the text) later when the sample is called out in the examples.